KALTURAPlayerSDK/GoogleCastProvider.m

```diff
@@ -223,19 +222,13 @@ - (void)setMediaSrc:(NSString *)mediaSrc {
 
 - (void)play {
     
-    if (_playerState != PlayerStatePlaying) {
+    if (_isEnded && _playerState != PlayerStatePlaying) {
         
-        if (_isEnded || _isChangeMedia) {
-            
-            NSString *metaData = [[NSUserDefaults standardUserDefaults] objectForKey: @"MetaDataCC"];
-            [self setVideoUrl:_mediaSrc startPosition:0 autoPlay:YES metaData: metaData];
-            _isEnded = NO;
-            _isChangeMedia = NO;
-        } else {
-            
-            [self setVideoUrl:_mediaSrc startPosition:0 autoPlay:YES metaData: nil];
```

Comment:
what about this case? when media is not ended and you try to change? did you test it?